File: src/main/java/com/viginfotech/chennaitimes/backend/utils/FPredicate.java
package com.viginfotech.chennaitimes.backend.utils;


import com.viginfotech.chennaitimes.backend.model.Feed;

import java.util.List;

/**
 * Created by anand on 2/1/16.
 */
public interface FPredicate<T> {
    boolean test(T type, Feed t);

    List<T> getTlistforThisCatagory(List<T> tlist, int category);
}

File: src/main/java/com/viginfotech/chennaitimes/backend/utils/FPredicate.kt
package com.viginfotech.chennaitimes.backend.utils


import com.viginfotech.chennaitimes.backend.model.Feed

/**
 * Created by anand on 2/1/16.
 */
interface FPredicate<T> {
    fun test(type: T, t: Feed): Boolean

    fun getTlistforThisCatagory(tlist: List<T>, category: Int): List<T>
}
